src/Build/Graph/ProjectGraph.cs

```diff
@@ -185,28 +185,52 @@ public ProjectGraph(IEnumerable<ProjectGraphEntryPoint> entryPoints, ProjectColl
                 entryPointConfigurationMetadata.Add(configurationMetadata);
             }
 
-            LoadGraph(projectsToEvaluate, projectCollection, tasksInProgress);
-            foreach(var configurationMetadata in entryPointConfigurationMetadata)
+            bool exceptionInLoading = false;
+            if (LoadGraph(projectsToEvaluate, projectCollection, tasksInProgress, out List<Exception> exceptions))
             {
-                entryPointNodes.Add(_allParsedProjects[configurationMetadata]);
-                if (!nodeStates.TryGetValue(_allParsedProjects[configurationMetadata], out var _))
+                foreach (var configurationMetadata in entryPointConfigurationMetadata)
                 {
-                    DetectCycles(_allParsedProjects[configurationMetadata], nodeStates, projectCollection, configurationMetadata.GlobalProperties);
+                    entryPointNodes.Add(_allParsedProjects[configurationMetadata]);
+                    if (!nodeStates.TryGetValue(_allParsedProjects[configurationMetadata], out var _))
+                    {
+                        try
```

Comment:
If `LoadGraph` returns true we don't have any exceptions right? Would it be worth just leaving `CircularDependencyException` a top-level exception here instead of wrapping it in the `AggregateException`?

IE, the ctor throws either a `CircularDependencyException` or a `AggregateException` containing invalid project exceptions

Reply:
sounds good.. will fix in the next iteration